cmd/interceptors/main.go

```diff
@@ -82,13 +111,108 @@ func main() {
 		IdleTimeout:  idleTimeout,
 		Handler:      mux,
 	}
-
-	logger.Infof("Listen and serve on port %d", Port)
-	if err := srv.ListenAndServe(); err != nil {
+	logger.Infof("Listen and serve on port %d", HTTPSPort)
+	if err := srv.ListenAndServeTLS(certFile, keyFile); err != nil {
 		logger.Fatalf("failed to start interceptors service: %v", err)
 	}
+
+}
+
+// updateCRDWithCaCert updates clusterinterceptor crd caBundle with caCert
+func updateCRDWithCaCert(ctx context.Context, cfg *rest.Config, caCert []byte) error {
+	tc, err := triggersclientset.NewForConfig(cfg)
+	if err != nil {
+		return err
+	}
+	clusterInterceptorList, err := tc.TriggersV1alpha1().ClusterInterceptors().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	for i := range clusterInterceptorList.Items {
+		if bytes.Equal(clusterInterceptorList.Items[i].Spec.ClientConfig.CaBundle, []byte{}) {
+			clusterInterceptorList.Items[i].Spec.ClientConfig.CaBundle = caCert
+			if _, err := tc.TriggersV1alpha1().ClusterInterceptors().Update(ctx, &clusterInterceptorList.Items[i], metav1.UpdateOptions{}); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
 }
 
 func handler(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 }
+
+// getCerts uses Knative pkg to generate certs for clusterinterceptor to run as https
+func getCerts(ctx context.Context, secretLister v1.SecretLister, kubeClient *kubeclientset.Clientset) (string, string, []byte, error) {
+	interceptorSvcName := os.Getenv("INTERCEPTOR_TLS_SVC_NAME")
+	interceptorSecretName := os.Getenv("INTERCEPTOR_TLS_SECRET_NAME")
+	namespace := os.Getenv("SYSTEM_NAMESPACE")
+
+	secret, err := secretLister.Secrets(namespace).Get(interceptorSecretName)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// The secret should be created explicitly by a higher-level system
+			// that's responsible for install/updates.  We simply populate the
+			// secret information.
+			log.Printf("secret %s is missing", interceptorSecretName)
+			return "", "", []byte{}, err
+		}
+		log.Printf("error accessing certificate secret %q: %v", interceptorSecretName, err)
+		return "", "", []byte{}, err
+	}
+
+	var (
+		serverKey, serverCert, caCert []byte
+		createCerts                   bool
+	)
+	serverKey, haskey := secret.Data[certresources.ServerKey]
+	if !haskey {
+		log.Printf("secret %q is missing key %q", secret.Name, certresources.ServerKey)
+		createCerts = true
+	}
+	serverCert, haskey = secret.Data[certresources.ServerCert]
+	if !haskey {
+		log.Printf("secret %q is missing key %q", secret.Name, certresources.ServerCert)
+		createCerts = true
+	}
+	caCert, haskey = secret.Data[certresources.CACert]
+	if !haskey {
+		log.Printf("secret %q is missing key %q", secret.Name, certresources.CACert)
+		createCerts = true
+	}
+
+	// TODO: Certification validation and rotation is pending
```

Comment:
@dibyom will create an issue to handle it in separate PR

Reply:
could you open an issue to track?